crates/bevy_asset/src/asset_server.rs

```diff
@@ -304,6 +304,16 @@ impl AssetServer {
         self.load_untyped(path).typed()
     }
 
+    /// Similar to [`AssetServer::load`] but this version will not panic. Instead, it will
+    /// emit an [`AssetServerError::IncorrectHandleType`] if there's a type mismatch.
+    #[must_use = "not using the returned strong handle may result in the unexpected release of the asset"]
+    pub fn try_load<'a, T: Asset, P: Into<AssetPath<'a>>>(
+        &self,
+        path: P,
+    ) -> Result<Handle<T>, AssetServerError> {
+        self.load_untyped(path).get_typed()
+    }
+
```

Comment:
this is not needed, it's not possible for `load` to be in the panicking case

Reply:
Interesting. I see:  I'd assumed we'd use the Uuid in both the "from file" and "runtime" cases.

Reply:
from file gives `AssetPath` ids. converting it to any typed handle will never fail because we actually have no idea of the asset type of the file, so we just accept everything

Reply:
😭 Okay, valid. So this method should be removed.